From e409b3a8ecabb054ee2ab3a141350102ec5346e4 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Mon, 19 Nov 2018 17:26:03 +0800
Subject: [PATCH 5160/5242] MLK-20207-1: input: synaptics_dsx_i2c: fix
 coverity wrong sizeof argument issue
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

commit  e8b0fe7d954a88b7d927b5af3ff9b797bded730b from
https://source.codeaurora.org/external/imx/linux-imx.git

CID 3869704: Wrong sizeof argument (SIZEOF_MISMATCH)
suspicious_sizeof: Passing argument f1a->button_control.txrx_map of
type unsigned char * and argument 8 /* sizeof (f1a->button_control.txrx_map) */
to function synaptics_rmi4_i2c_read is suspicious.

Passing argument should be sizeof(unsigned char) rather than
sizeof(unsigned char *).

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
(cherry picked from commit 4583a0bc1ff6684e1d850eb50308ad8ae2fea627)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../touchscreen/synaptics_dsx/synaptics_dsx_i2c.c  |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/synaptics_dsx/synaptics_dsx_i2c.c b/drivers/input/touchscreen/synaptics_dsx/synaptics_dsx_i2c.c
index 1ecf79c..64089d3 100644
--- a/drivers/input/touchscreen/synaptics_dsx/synaptics_dsx_i2c.c
+++ b/drivers/input/touchscreen/synaptics_dsx/synaptics_dsx_i2c.c
@@ -2229,7 +2229,7 @@ static int synaptics_rmi4_f1a_button_map(struct synaptics_rmi4_data *rmi4_data,
 		retval = synaptics_rmi4_i2c_read(rmi4_data,
 				fhandler->full_addr.ctrl_base + mapping_offset,
 				f1a->button_control.txrx_map,
-				sizeof(f1a->button_control.txrx_map));
+				sizeof(*(f1a->button_control.txrx_map)));
 		if (retval < 0) {
 			dev_err(&rmi4_data->i2c_client->dev,
 					"%s: Failed to read tx rx mapping\n",
-- 
1.7.9.5

